test/TestNestedJinjaRule.py

```diff
@@ -0,0 +1,190 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import os
+import pytest
+from collections import namedtuple
+from ansiblelint import Runner, RulesCollection
+
+PlayFile = namedtuple('PlayFile', ['name', 'content'])
+
+FAIL_TASK_1LN = PlayFile('playbook.yml', u'''
+- name: one-level nesting
+  set_fact:
+    var_one: "2*(1+2) is {{ 2 * {{ 1 + 2 }} }}"
+''')
+
+FAIL_TASK_1LN_M = PlayFile('playbook.yml', u'''
+- name: one-level multiline nesting
+  set_fact:
+    var_one_ml: >
+      2*(1+2) is {{ 2 *
+      {{ 1 + 2 }}
+      }}
+''')
+
+FAIL_TASK_2LN = PlayFile('playbook.yml', u'''
+- name: two-level nesting
+  set_fact:
+    var_two: "2*(1+(3-1)) is {{ 2 * {{ 1 + {{ 3 - 1 }} }} }}"
+''')
+
+FAIL_TASK_2LN_M = PlayFile('playbook.yml', u'''
+- name: two-level multiline nesting
+  set_fact:
+    var_two_ml: >
+      2*(1+(3-1)) is {{ 2 *
+      {{ 1 +
+      {{ 3 - 1 }}
+      }} }}
+''')
+
+FAIL_TASK_W_5LN = PlayFile('playbook.yml', u'''
+- name: five-level wild nesting
+  set_fact:
+    var_three_wld: "{{ {{ {{ {{ {{ 234 }} }} }} }} }}"
+''')
+
+FAIL_TASK_W_5LN_M = PlayFile('playbook.yml', u'''
+- name: five-level wild multiline nesting
+  set_fact:
+    var_three_wld_ml: >
+      {{
+      {{
+      {{
+      {{
+      {{ 234 }}
+      }}
+      }}
+      }}
+      }}
+''')
+
+SUCCESS_TASK_P = PlayFile('playbook.yml', u'''
+- name: proper non-nested example
+  set_fact:
+    var_one: "number for 'one' is {{ 2 * 1 }}"
+''')
+
+SUCCESS_TASK_P_M = PlayFile('playbook.yml', u'''
+- name: proper multiline non-nested example
+  set_fact:
+    var_one_ml: >
+      number for 'one' is {{
+      2 * 1 }}
+''')
+
+SUCCESS_TASK_2P = PlayFile('playbook.yml', u'''
+- name: proper nesting far from each other
+  set_fact:
+    var_two: "number for 'two' is {{ 2 * 1 }} and number for 'three' is {{ 4 - 1 }}"
+''')
+
+SUCCESS_TASK_2P_M = PlayFile('playbook.yml', u'''
+- name: proper multiline nesting far from each other
+  set_fact:
+    var_two_ml: >
+      number for 'two' is {{ 2 * 1
+      }} and number for 'three' is {{
+      4 - 1 }}
+''')
+
+SUCCESS_TASK_C_2P = PlayFile('playbook.yml', u'''
+- name: proper nesting close to each other
+  set_fact:
+    var_three: "number for 'ten' is {{ 2 - 1 }}{{ 3 - 3 }}"
+''')
+
+SUCCESS_TASK_C_2P_M = PlayFile('playbook.yml', u'''
+- name: proper multiline nesting close to each other
+  set_fact:
+    var_three_ml: >
+      number for 'ten' is {{
+      2 - 1
+      }}{{ 3 - 3 }}
+''')
+
+@pytest.fixture
+def play_file_path(tmp_path):
+    p = tmp_path / 'playbook.yml'
+    return str(p)
+
+
+@pytest.fixture
+def rules():
+    rulesdir = os.path.join('lib', 'ansiblelint', 'rules')
+    return RulesCollection([rulesdir])
+
+
+@pytest.fixture
+def runner(play_file_path, rules):
+    return Runner(rules, play_file_path, [], [], [])
```

Comment:
There's now `default_rules_collection` fixture in `conftest`. So you don't really need `play_file_path` and `rules` fixtures anymore:
```suggestion
def runner(tmp_path, default_rules_collection):
    return Runner(
        default_rules_collection,
        str(tmp_path / 'playbook.yml'),
        [], [], [],
    )
```
like @ https://github.com/ansible/ansible-lint/commit/6702769f4488529e16dc5f3dc320a363586364eb#diff-e21acb2ef339763de8d17d0a8be0f295R33-R34

Reply:
Yes, you are right https://github.com/ansible/ansible-lint/blob/master/test/conftest.py#L10

```py
@pytest.fixture
@pytest.mark.usefixtures('default_rules_collection')
def runner(tmp_path, default_rules_collection):
    return Runner(
        default_rules_collection,
        str(tmp_path / 'playbook.yml'),
        [], [], [],
    )
```
```
file /home/adrian/Documents/ansible-lint/test/TestNestedJinjaRule.py, line 129                                                                
  @pytest.fixture                                                                                                                             
  def runner(tmp_path, default_rules_collection):                                                                                             
E       fixture 'default_rules_collection' not found                                                                                          
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, monkeypatch, no_cover, play_files
, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, runner, testrun_uid, tmp_path, tmp_path_factory, tm
pdir, tmpdir_factory, worker_id
```

I tried to define a fixture using scope too as it should be, but I did not succeed.

Reply:
I see it works now. I guess you forgot to rebase, right?

Reply:
Yes. It was due to outdated changes. I have already rebased my branch against upstream/master.